### DISCUSSION TITLE
Github Actions Trigger not working

### DISCUSSION THREAD

**Discussion by ramandatascientist** [community/community#32955] [Actions]
Hello Folks, 

We have a Github actions job that listens for any push on the following branches and trigger this workflow if there is any push

- dev/collector
- test/collector
- production/collector

On our main/ branch, we have series of commits (eg: A --> B--> C), so this above workflow gets trigger if i deploy these commits to dev/collector (example)

```
A --> B
B --> C
```

But it does not trigger when we do backwards (eg:

` C-->B, B---> A`

Can someone please advise why this behaviour is happening? is that a bug?

<img width="553" alt="image" src="https://user-images.githubusercontent.com/50452830/190001560-750a6cbc-8c69-45c5-a073-ed6ec2ee74a3.png">

**Comment by queenofcorgis** [comment]
Welcome to the GitHub Community, @ramandatascientist! We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category. I've gone ahead and moved it for you. Good luck!

**Reply by queenofcorgis** [reply]
Also, as a reminder, the GitHub Community on Discussions is a space for all GitHub users to come and discuss ideas and projects. Your post was removed because it contained content that violates our Community Code of Conduct, including posting the same comment in multiple places. As a result, we have deleted the duplicate comments, and we ask that you no longer cross-post.

For additional guidance, please review our [Community on Discussions Code of Conduct](https://github.com/community/community/blob/main/CODE_OF_CONDUCT.md) and [GitHub’s Terms of Service](https://docs.github.com/en/github/site-policy/github-terms-of-service).

**Comment by airtower-luna** [comment]
> On our main/ branch, we have series of commits (eg: A --> B--> C), so this above workflow gets trigger if i deploy these commits to dev/collector (example)

What exactly do you mean by that? "Deploying commits" is not a Git term, maybe you can describe what *exactly* (commands, etc.) you do? And then the same thing for the different order. Git commits have a fixed order in history, unless you rewrite history you can't push them in a different order.

**Reply by ramandatascientist** [reply]
Hi @airtower-luna , I am adding more details on how i am reproducing the issue:


We have a Github actions job that listens for any push on the following branches and trigger this workflow if there is any push

```
dev/collector
test/collector
production/collector
```
![image](https://user-images.githubusercontent.com/50452830/190014917-bf3a3234-2962-4a55-9dea-9444f4b57cf5.png)

On our main/ branch, we have a series of commits (eg:` A --> B--> C`) and each commit is tagged as well

eg: 

tag-A (for commit A)
tag-B (for commit B) 
tag-C (for commit B)  


so this above workflow gets triggered if i repoint the HEAD of dev/collector in the following sequence(example)

Step 1: A --> B (command: `git push origin Tag-B:test/abc --force`) (it triggers the workflow as expected)
Step 2: B --> C (command: `git push origin Tag-C:test/abc --force` (it triggers the workflow as expected)
Step 3: C --> B (command: `git push origin Tag-B:test/abc --force` (it does not trigger the workflow and thats where i see the issue)

**Reply by airtower-luna** [reply]
I see what you mean, and that is odd. I wonder if there's some confusion because there was already a push event for that commit on that branch? :thinking: You could try to [send a bug report to support](https://support.github.com/contact/bug-report) to see if that's intentional.

However, I can't help but wonder: Why would you do that? :sweat_smile:

**Reply by ramandatascientist** [reply]
to allow rollbacks (eg: if C is not working then re-point the branch (dev/collector to Commit B)

**Reply by airtower-luna** [reply]
Ah, I see. In that case you could try reverting commit C instead to work around the problem: That adds a new commit that undos the changes, so there is a new commit being pushed.

**Reply by ramandatascientist** [reply]
actually we like to build a job that allows users to enter any tags (tag-a, tag-b, tag-c) without typing any github commands

<img width="365" alt="image" src="https://user-images.githubusercontent.com/50452830/190162399-e2f31b69-c825-4388-aee3-bb6095c6cc00.png">

so in that case, how can i add that 'revert' logic inside our workflow?

**Reply by airtower-luna** [reply]
If you already have the user enter which tag they'd like to deploy, why not check out that tag and deploy from there, regardless of any branch?

**Reply by ramandatascientist** [reply]
hmm not sure if i understand that. Can you share some commands and examples (specific to my use case)?

**Reply by airtower-luna** [reply]
Not in detail without knowing your actual workflow. But maybe I can be a bit more specific about the general idea.

* You have a tag you want to deploy from (entered by the user).
* As far as I understand that tag is currently force-pushed to a branch, which (should) then trigger a deployment from that branch. Remove that.
* Somewhere in your workflow you check out the repository (usually using [`actions/checkout`](https://github.com/actions/checkout)). At that point you can choose which `ref` (branch, tag, commit, etc.) to check out. Check out the selected tag.
* Deploy based on the checked out data (which is whatever the repository contains at the tag).

**Reply by ramandatascientist** [reply]
Thank for the details. I am sharing more details on our workflow here: https://support.github.com/ticket/personal/0/1793973

can you help me identify where to make that change and how?

**Reply by airtower-luna** [reply]
Support tickets are only visible to GitHub staff. If it's not public I can't help.
